packages/effect/src/Micro.ts

```diff
@@ -3671,7 +3741,14 @@ export interface Handle<A, E = never> {
 export const isHandle = (u: unknown): u is Handle<unknown, unknown> =>
   typeof u === "object" && u !== null && HandleTypeId in u
 
-class HandleImpl<A, E> implements Handle<A, E> {
+class HandleImpl<A, E> extends MicroableClass<A, E> implements Handle<A, E> {
+  readonly [runSymbol] = (_env: Env<any>, onExit: (exit: MicroExit<A, E>) => void) => {
+    const observer = (exit: MicroExit<A, E>) => {
```

Comment:
This should handle interruption too.

Reply:
Done, but please check it again)

Reply:
I added an abstract `asMicro` method to the Class, as implementing the `runSymbol` function has a bunch of footguns that you have to be aware of.